plugins/modules/sda_fabric_devices_workflow_manager.py

```diff
@@ -2599,27 +2604,55 @@ def get_device_params(self, fabric_id, network_id, device_details, config_index)
                     )
                     self.status = "failed"
                     return self.check_return_status()
+            else:
+                if have_layer3_settings and (str(local_autonomous_system_number) != str(have_layer3_settings.get("localAutonomousSystemNumber"))):
```

Comment:
existing_as_number = have_layer3_settings.get("localAutonomousSystemNumber") if have_layer3_settings else None
    if existing_as_number and str(local_autonomous_system_number) != str(existing_as_number):
        self.msg = (
            "The parameter 'local_autonomous_system_number' in 'layer3_settings' must not be updated "
            "for the device with IP '{ip}'.".format(ip=device_ip)
        )
        self.status = "failed"
        return self.check_return_status()